DEVELOPERS_README.dox

```diff
@@ -0,0 +1,180 @@
+
+/*! \page VOTCA Developers Guide
+ *
+ * The page is designed to give new developers general guidelines for
+ * implementing code consistent with the VOTCA and c++ style and standard.
+ *
+ *  \section General Resources
+ *
+ * A good starting point is to take a look at the c++ standard. Though the code
+ * has not always consistently followed the c++ standard we have made a recent
+ * effort to really enforce it and follow best practices.
+ *
+ * - https://www.gitbook.com/book/lefticus/cpp-best-practices/details
+ * - https://google.github.io/styleguide/cppguide.html
+ *
+ * \section Testing
+ *
+ * Each repository contains a src folder. Within the src folder exists a 
+ * libraryfolder: libtools, libcsg etc... and a tools folder. A tests folder
+ * should also exist in the src folder. If it does not you should create one.
+ *
+ * For every new object and algorithm created there should exist a test. We
+ * will be using the Boost libraries testing framework. Good documentation can
+ * be found here:
+ *
+ * - https://www.ibm.com/developerworks/aix/library/au-ctools1_boost/
+ * 
+ * We will outline the general workflow here using the vec object in 
+ * votca::tools. This object only has a header file it is in:
+ * /tools/include/votca/tools/vec.h
+ * 
+ * Determine if a tests folder has already been created or not in /src if it
+ * has we will skip to step 5
```

Comment:
I am not sure if we should be that explicit, extensive documentation tends to get never updated and outdated very quickly. I would just say to look at the setup in votca-tools as an example.

Ultimately all votca repos will hopefully have a test folder in the near future.